osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs

```diff
@@ -398,38 +397,7 @@ private void onRoomUpdated()
         private void updateCurrentItem()
         {
             Debug.Assert(client.Room != null);
-
-            var expectedSelectedItem = Room.Playlist.SingleOrDefault(i => i.ID == client.Room.Settings.PlaylistItemId);
-
-            if (expectedSelectedItem == null)
-                return;
-
-            // There's no reason to renew the selected item if its content hasn't changed.
-            if (SelectedItem.Value?.Equals(expectedSelectedItem) == true && expectedSelectedItem.Beatmap.Value != null)
-                return;
-
-            // Clear the selected item while the lookup is performed, so components like the ready button can enter their disabled states.
-            SelectedItem.Value = null;
-
-            if (expectedSelectedItem.Beatmap.Value == null)
-            {
-                Task.Run(async () =>
-                {
-                    var beatmap = await client.GetAPIBeatmap(expectedSelectedItem.BeatmapID).ConfigureAwait(false);
-
-                    Schedule(() =>
-                    {
-                        expectedSelectedItem.Beatmap.Value = beatmap;
-
-                        if (Room.Playlist.SingleOrDefault(i => i.ID == client.Room?.Settings.PlaylistItemId)?.Equals(expectedSelectedItem) == true)
-                            applyCurrentItem();
-                    });
-                });
-            }
-            else
-                applyCurrentItem();
-
-            void applyCurrentItem() => SelectedItem.Value = expectedSelectedItem;
+            SelectedItem.Value = Room.Playlist.SingleOrDefault(i => i.ID == client.Room.Settings.PlaylistItemId);
```

Comment:
To clarify this, because it wasn't spelled out explicitly - this (along with other similar changes) is safe to do, because `MultiplayerClient.createPlaylistItem()` will initially create `PlaylistItem`s with a "shell" `APIBeatmap` that only has the online ID populated? And all of the other info is fetched asynchronously from API, and once that's done, the "shell" models are replaced with fully populated ones in the `Playlist` - do I have that right?

Reply:
Partly correct. The shell model is as you described, however it's never updated to the full beatmap. Every component which requires the full beatmap does a lookup via the `BeatmapLookupCache`.

This particular change is fine to do now because the relevant components (`OnlinePlayBeatmapAvailabilityTracker`) do their own lookups internally.